### PR TITLE
[Team3] #57 #71 Add Powertrain Outbound Interfaces + Vector Implementation

### PR DESCRIPTION
**Pull request by AbyssPrince**
## Definition of Done
- [x] add enum for transmission mode (`CarTransmissionMode` enum)
- [x] add enum `Axis` (`X`, `Y`)
- [x] implement `IVector` class
- [x] add packet interface for car position and movement (`CarPositionPacket` interface)
- [x] add packet interface for engine status (`EngineStatusPacket` interface)
- [x] add data implementation classes for mentioned interfaces (same names with `Data` suffix)
- [x] share interfaces of packets on virtual bus publicly

### CODE REVIEW

File: src/main/java/hu/oe/nik/szfmv/automatedcar/powertrain/CarTransmissionMode.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    N_NEUTRAL,
+
+    /**Parkoló mód.*/
+    P_PARKING,
+
+    /**Autómata állás.*/
+    D_DRIVE,
+
+    /**Manuális állás.*/
+    F_FORWARD
```
Comment (Szajkop) [inline src/main/java/hu/oe/nik/szfmv/automatedcar/powertrain/CarTransmissionMode.java]: A "manuális állással" nem kell foglalkozni

Comment (Keev2n) [reply src/main/java/hu/oe/nik/szfmv/automatedcar/powertrain/CarTransmissionMode.java]: +1

Comment (varszegimarcell) [reply src/main/java/hu/oe/nik/szfmv/automatedcar/powertrain/CarTransmissionMode.java]: ++

**Comment by AbyssPrince** [comment]
@pintergreg Hi there,
can I ask how can I refresh the pipeline results with coverages, etc... here? I added unit tests so I expect the coverage to improve. The coverage report visible above does not contain any useful information. Thanks in advance!

**Comment by pintergreg** [comment]
@AbyssPrince Egyelőre nem tudom. Első nekifutásra nem sikerült megoldani.

**Pull request merged**